modules/dnn/src/layers/layers_common.cpp

```diff
@@ -188,42 +202,44 @@ void getConvolutionKernelParams(const LayerParams &params, int &kernelH, int &ke
 // We pad Pr/2 on the left and Pr - Pr/2 on the right, Pc/2 on the top
 // and Pc - Pc/2 on the bottom.  When Pr or Pc is odd, this means
 // we pad more on the right and bottom than on the top and left.
-void getConvPoolOutParams(const Size& inp, const Size &kernel,
-                          const Size &stride, const String &padMode,
-                          const Size &dilation, Size& out)
+void getConvPoolOutParams(const std::vector<int>& inp, const std::vector<size_t>& kernel,
+                          const std::vector<size_t>& stride, const String &padMode,
+                          const std::vector<size_t>& dilation, std::vector<int>& out)
 {
     if (padMode == "VALID")
     {
-        out.height = (inp.height - (dilation.height * (kernel.height - 1) + 1) + stride.height) / stride.height;
-        out.width = (inp.width - (dilation.width * (kernel.width - 1) + 1) + stride.width) / stride.width;
+        for (int i = 0; i < inp.size(); i++)
+            out.push_back((inp[i] - dilation[i] * (kernel[i] - 1) - 1 + stride[i]) / stride[i]);
     }
     else if (padMode == "SAME")
     {
-        out.height = (inp.height - 1 + stride.height) / stride.height;
-        out.width = (inp.width - 1 + stride.width) / stride.width;
+        for (int i = 0; i < inp.size(); i++)
+            out.push_back((inp[i] - 1 + stride[i]) / stride[i]);
     }
     else
-    {
         CV_Error(Error::StsError, "Unsupported padding mode");
-    }
 }
 
-void getConvPoolPaddings(const Size& inp, const Size& out,
-                         const Size &kernel, const Size &stride,
-                         const String &padMode, const Size &dilation, int &padT, int &padL, int &padB, int &padR)
+void getConvPoolPaddings(const std::vector<int>& inp, const std::vector<int>& out,
+                         const std::vector<size_t>& kernel, const std::vector<size_t>& strides,
+                         const String &padMode, const std::vector<size_t>& dilation,
+                         std::vector<size_t>& pads_begin, std::vector<size_t>& pads_end)
 {
+    CV_Assert_N(kernel.size() == pads_begin.size() || pads_begin.size() == 1, pads_begin.size() == pads_end.size());
+    pads_begin.resize(kernel.size(), pads_begin[0]);
+    pads_end.resize(kernel.size(), pads_end[0]);
+    CV_Assert(pads_begin.size() == pads_end.size() && pads_begin.size() == kernel.size());
```

Comment:
It seems to me that at this point `kernel.size()` must be equal to `pads_begin.size()` and pads_end.size()`. May you check this?